File: MBStore/Models/Product.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace MBStore.Models
{
    public class Product
    {
        public Guid ProductId { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string ImageUrl { get; set; }
        public decimal ExcludingTaxPrice { get; set; }
        public decimal IncludingTaxPrice { get; set; }
        public decimal VATRate { get; set; }

        public IList<CartItemProduct> CartItemProducts { get; set; }
    }
}

File: MBStore/ClientApp/src/app/shared/models/Product.ts
import { CartItemProduct } from './CartItemProduct';

export class Product {
    public productId: string;
    public name: string;
    public description: string;
    public imageUrl: string;
    public excludingTaxPrice: number;
    public includingTaxPrice: number;
    public vatRate: number;
    public cartItemProducts: CartItemProduct[];
}
